JavaScript/JavaScript.sublime-syntax

```diff
@@ -162,11 +170,54 @@ contexts:
     - match: (?=\S)|$
       pop: 1
 
-  comments-top-level:
-    - match: ^(#!).*$\n?
-      scope: comment.line.shebang.js
+  line-comments:
+    - match: /{4,}
+      scope: punctuation.definition.comment.js
+      push: line-comment-other-body
+    - match: /{3}
+      scope: punctuation.definition.comment.js
+      push: line-comment-triple-slash-body
+    - match: /{2}
+      scope: punctuation.definition.comment.js
+      push: line-comment-double-slash-body
+
+  line-comment-end:
+    - match: (//+)?\n
```

Comment:
```suggestion
    - match: (?://+)?\n
```

Reply:
1. I am not sure about whether we should scope the newline character `punctuation`
2. This change would break syntax due to following `caputures 1`

Reply:
Oops, my bad.